client/dashboard/sites/fields.tsx

```diff
@@ -27,14 +30,42 @@ function getSiteManagementUrl( site: Site ) {
 	return site.options?.admin_url;
 }
 
+function SiteBadge( { site }: { site: Site } ) {
+	if ( site.is_wpcom_staging_site ) {
+		return <Badge>{ __( 'Staging' ) }</Badge>;
+	}
+
+	if ( isSitePlanTrial( site ) ) {
+		return <Badge>{ __( 'Trial' ) }</Badge>;
+	}
+
+	if ( isP2( site ) ) {
+		return <Badge>{ __( 'P2' ) }</Badge>;
+	}
+
+	return null;
+}
+
 const DEFAULT_FIELDS: Field< Site >[] = [
 	{
 		id: 'name',
 		label: __( 'Site' ),
 		enableGlobalSearch: true,
 		getValue: ( { item } ) => item.name || new URL( item.URL ).hostname,
 		render: ( { field, item } ) => (
-			<Link to={ getSiteManagementUrl( item ) }>{ field.getValue( { item } ) }</Link>
+			<Link to={ getSiteManagementUrl( item ) }>
+				<HStack alignment="center" spacing={ 1 }>
+					<Text
+						as="span"
+						style={ { overflowX: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap' } }
+					>
+						{ field.getValue( { item } ) }
+					</Text>
+					<Text as="span" style={ { flexShrink: 0 } }>
+						{ SiteBadge( { site: item } ) }
```

Comment:
Looks weird. Should it be `<SiteBadge site={ item } />`?

Reply:
df90fa2384049d4062a13f866d9e7aa1accb777e